test_data/asm/trait_parsing.asm

```diff
@@ -0,0 +1,13 @@
+machine Trait with degree: 8 {
+    reg pc[@pc];
+    reg X[<=];
+    reg A;
+
+    trait ArraySum<T> {
```

Comment:
Oh you don't really need machines here for a test of the asm part. I would actually consider not allowing trait declarations inside machines if that makes it easier somehow (maybe it will have to later).